src/pfe/portal/modules/User.js

```diff
@@ -922,9 +959,15 @@ module.exports = class User {
       return registrySecretList;
     }
 
-    const msg = "Failed to patch the Service Account";
+    const msg = "Reverting changes to the Docker Config";
     log.error(msg);
-    throw new RegistrySecretsError("SERVICE_ACCOUNT_PATCH_FAILED", "for address " + address);
+
+    // Since patching the Service Account failed, we need to revert the delete from the Docker Config and patch the Service Account again
+    jsonObj.auths[address] = registrySecretToBeDeleted;
+    await fs.writeJson(this.dockerConfigFile, jsonObj);
+    await this.updateServiceAccountWithDockerRegisrySecret();
```

Comment:
I might be missing something kubernetes related here but if we fail to patch the service account:
- Do we need to repatch it back to the old settings? Won't they still be set?
- Will we be able to change them? The patch operation just failed once, why will it work this time?

Reply:
1. It may or may not, I think in the case of a failed API operation, since we're rolling back to the previous state, does it not make sense to have the secret re-created with the prev docker config and patch the SA?

2. Yes it may have failed in the API operation(could be secret creation failed or patching SA failed), but we should at least attempt to restore it back to what it was.

What do you think?

Reply:
Actually, I see Howard's point. If the patch fails then the only the docker config file and secret need to be rolled back. What we have should be harmless though because the patch to roll back should just fail and if it somehow works it'll match up with the secret.